### DISCUSSION TITLE
Can I pass back data to the python code calling pytest.main() where pytest.main() is executing unittests.

### DISCUSSION THREAD

**Discussion by PatrickFlaherty** [pytest-dev/pytest#11505] [Q&A]
Can I pass back data to the python code calling pytest.main() where pytest.main() is executing unittests.
`class AllureRunner:

    def run_tests_with_allure_report(self, tests_path):

        params = ['--tb=native',  f'--alluredir=allure_dir\\results', f'{tests_path}']

        # Run Unit Tests
        response = pytest.main(params)

        print("return code from calling pytest.main():", response)`

**Comment by RonnyPfannschmidt** [comment]
The second argument to the main function is a list of pytest plugin objects each of which can implement any pytest Hook

**Comment by PatrickFlaherty** [comment]
@RonnyPfannschmidt thanks for the response. I have not used Hooks before, Could you provide an example of this?
Thank you!

**Reply by RonnyPfannschmidt** [reply]
I believe there's one in the docs,in case you don't find it please Ping me so I can find it when I get back to the computer tomorrow

**Comment by PatrickFlaherty** [comment]
I will have a look, thank you @RonnyPfannschmidt .

**Reply by PatrickFlaherty** [reply]
Hi @RonnyPfannschmidt , have not found that example, would you be so kind as to post if you have a moment. Thank you.

**Reply by RonnyPfannschmidt** [reply]
https://docs.pytest.org/en/7.1.x/example/simple.html?highlight=main#freezing-pytest has a example of passing a plugin object (in this case its a python module with the hooks

https://github.com/pytest-dev/pytest/blob/cdddd6d69533c07092c64445a0d42d1fc75c8299/src/_pytest/junitxml.py#L464-L700 is a example of a plugin object, that when registered with pytest makes the junitxml report

**Comment by PatrickFlaherty** [comment]
Thank you @RonnyPfannschmidt !
